.github/workflows/main.yml

```diff
@@ -0,0 +1,68 @@
+name: Test & release
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - main
+
+jobs:
+  test:
+    name: npm test
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        node: [16]
+    steps:
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: "npm"
+
+      - name: 📥 Download deps
+        run: npm install
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: ▶️ Run tests
+        run: npm run test
+
+  release:
+    name: 🚀 Release
+    needs: [test]
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.fork &&
+      contains('refs/heads/master,refs/heads/main', github.ref) && github.event_name == 'push' }}
+    steps:
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: 📥 Download deps
+        run: npm install
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: 🏗 Run build script
+        run: npm run --if-present build
+
+      - name: 🚀 Release
+        uses: tradeshift/actions-semantic-release@v1
+        id: semantic-release
+        with:
+          registry: https://npm.pkg.github.com/
+          npm_publish: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Ah, we need to use the public registry for this public package.

https://www.npmjs.com/package/http-mockserver

See https://github.com/Tradeshift/eslint-config-tradeshift/blob/58b999f5/.github/workflows/main.yml#L63-L79 for example.

Reply:
good catch :)